src/symbol/sprite_atlas.js

```diff
@@ -3,24 +3,20 @@
 const ShelfPack = require('@mapbox/shelf-pack');
 const browser = require('../util/browser');
 const util = require('../util/util');
+const window = require('../util/window');
+const Evented = require('../util/evented');
 
-class AtlasImage {
-    constructor(rect, width, height, sdf, pixelRatio) {
-        this.rect = rect;
-        this.width = width;
-        this.height = height;
-        this.sdf = sdf;
-        this.pixelRatio = pixelRatio;
-    }
-}
-
-class SpriteAtlas {
+// The SpriteAtlas class is responsible for turning a sprite and assorted
+// other images added at runtime into a texture that can be consumed by WebGL.
+class SpriteAtlas extends Evented {
 
     constructor(width, height) {
+        super();
+
         this.width = width;
         this.height = height;
 
-        this.atlas = new ShelfPack(width, height);
+        this.shelfPack = new ShelfPack(width, height);
```

Comment:
I'm not sure about this rename. `ShelfPack` is more of an algorithm name than a name that describes an instance. `atlas` is better IMO, and consistent with the class name (`SpriteAtlas`) since the latter is a wrapper around a generic atlas for image sprites.